webui/src/pages/repositories/repository/changes.jsx

```diff
@@ -251,50 +259,13 @@ const ChangesBrowser = ({repo, reference, prefix, onSelectRef, }) => {
-                                <URINavigator path={prefix} reference={reference} repo={repo}
-                                              pathURLBuilder={(params, query) => {
-                                                  return {
-                                                      pathname: '/repositories/:repoId/changes',
-                                                      params: params,
-                                                      query: {ref: reference.id, prefix: query.path ?? ""},
-                                                  }
-                                              }}/>
-                            )}
-                        </span>
-                        </Card.Header>
-                        <Card.Body>
-                            <Table borderless size="sm">
-                                <tbody>
-                                {results.map(entry => {
-                                    const committedRef = reference.id + "@"
-                                    const uncommittedRef = reference.id
-                                    return (
-                                        <TreeItem key={entry.path + "-tree-item"} entry={entry} repo={repo}
-                                                  reference={reference}
-                                                  leftDiffRefID={committedRef} rightDiffRefID={uncommittedRef}
-                                                  internalReferesh={internalRefresh}
-                                                  onNavigate={onNavigate} onRevert={onRevert} delimiter={delimiter}
-                                                  relativeTo={prefix}
-                                                  getMore={(afterUpdated, path, useDelimiter= true, amount = -1) => {
-                                                      return refs.changes(repo.id, reference.id, afterUpdated, path, useDelimiter ? delimiter : "", amount > 0 ? amount : undefined)
-                                                  }}/>
-                                        )
-                                })}
-                                { !!nextPage &&
-                                    <TreeEntryPaginator path={""} loading={loading} nextPage={nextPage} setAfterUpdated={setAfterUpdated}/>
-                                }
-                                </tbody>
-                            </Table>
-                        </Card.Body>
-                    </Card>
-                )}
-            </div>
+            <ChangesTreeContainer results={results} delimiter={delimiter}
+                                  uriNavigator={uriNavigator} leftDiffRefID={committedRef} rightDiffRefID={uncommittedRef}
+                                  repo={repo} reference={reference} internalReferesh={internalRefresh} prefix={prefix}
+                                  getMore={(afterUpdated, path, useDelimiter= true, amount = -1) => {
```

Comment:
Make the anonymous function passed to the `getMore` prop a named function, stop with the ternary ifs everywhere (!!), and `useCallback` as needed.

Reply:
Since you're re-using the function, move it to the same place where `ChangesTreeContainer` is and provide that default implementation as a default prop value. If another implementation is needed, it can always be overwritten.
If you want the consumer to provide `refs`, you can do something like this
```JavaScript
export const defaultGetMore = (refs) => 
  (afterUpdated, path, useDelimiter= true, amount = -1) => { ... }; 
```
and you can use `refs` from the closure <-- functional programming pro tip 😁

Reply:
done, but - 
- I think that ternary ifs are fine here as they make things short
- I don't see how I can use useCallback here because it will make me violate hooks rules. 

let me know if the current implementation looks ok or still create issues